docs/local-setup.md

```diff
@@ -0,0 +1,374 @@
+
+- [GitHub Repository](https://github.com/deepnote/jupyterlab-deepnote)
+- [PyPI Package](https://pypi.org/project/jupyterlab-deepnote/)
+- [Contributing Guide](https://github.com/deepnote/jupyterlab-deepnote/blob/main/CONTRIBUTING.md)
+
+## Deepnote Toolkit (advanced)
+
+The **Deepnote Toolkit** is a Python package that provides the underlying infrastructure for running Deepnote notebooks locally. This is an advanced option for developers who want to build custom implementations or deeply integrate Deepnote into their workflows.
+
+### What is the Deepnote Toolkit?
+
+The toolkit is the core Python package that powers Deepnote's execution environment. It includes:
+
+- **Python kernel** with scientific computing libraries
+- **SQL support** with query caching
+- **Data visualization** (Altair, Plotly)
+- **Streamlit apps** support with auto-reload
+- **Language Server Protocol** integration
+- **Git integration** with SSH/HTTPS authentication
+- **Integration environment variables** management
+
+### When to use the Toolkit
+
+Consider using the Deepnote Toolkit if you:
+
+- ✅ Want to build custom notebook execution environments
+- ✅ Need to integrate Deepnote notebooks into existing Python applications
+- ✅ Are developing extensions or tools for Deepnote
+- ✅ Want full control over the execution environment
+- ✅ Need to customize the notebook runtime behavior
+
+### Installation
+
+**Requirements:**
+
+- Python 3.10+
+- Poetry (for dependency management)
+- Java 11 (for PySpark features)
+
+**Install the toolkit:**
+
+```bash
+# Install with pip
+pip install deepnote-toolkit
+
+# Or with Poetry
+poetry add deepnote-toolkit
+```
+
+### CLI quick start
+
+The toolkit includes a command-line interface for running Jupyter servers:
+
+```bash
+# Start Jupyter server on default port (8888)
+deepnote-toolkit server
+
+# Start with custom configuration
+deepnote-toolkit server --jupyter-port 9000
+
+# View configuration
+deepnote-toolkit config show
+
+# Modify configuration
+deepnote-toolkit config set server.jupyter_port 9000
+```
+
+**Security Warning**: The CLI will warn if Jupyter runs without authentication. This is intended for local development only. Set `DEEPNOTE_JUPYTER_TOKEN` for shared environments.
+
+### Using in Python Code
+
+```python
+from deepnote_toolkit import DeepnoteKernel
+from deepnote_toolkit.sql import execute_sql
+
+# Execute SQL queries
+result = execute_sql(
+    query="SELECT * FROM users LIMIT 10",
+    connection_string="postgresql://localhost/mydb"
+)
+
+# Use Deepnote components
+from deepnote_toolkit.components import DataTable
+DataTable(result)
+```
+
+### Development setup
+
+For developers who want to contribute or customize the toolkit:
+
+```bash
+# Clone the repository
+git clone https://github.com/deepnote/deepnote-toolkit.git
+cd deepnote-toolkit
+
+# Install dependencies
+poetry install
+
+# Run tests
+poetry run nox -s unit
+
+# Start development server
+poetry run deepnote-toolkit server
+```
```

Comment:
Imo not very relevant for docs on deepnote.com, we should remove this section and link to the corresponding CONTRIBUTING.md at most.